src/csharp/Grpc.Core/Internal/UnmanagedLibrary.cs

```diff
@@ -82,6 +82,32 @@ public UnmanagedLibrary(string[] libraryPathAlternatives)
         /// <returns></returns>
         public IntPtr LoadSymbol(string symbolName)
         {
+            if (PlatformApis.IsWindows)
+            {
+                // See http://stackoverflow.com/questions/10473310 for background on this.
+                if (PlatformApis.Is64Bit)
+                {
+                    return Windows.GetProcAddress(this.handle, symbolName);
+                }
+                else
+                {
+                    // Yes, we could potentially predict the size... but it's a lot simpler to just try
+                    // all the candidates. Most functions have a suffix of @0, @4 or @8 so we won't be trying
+                    // many options - and if it takes a little bit longer to fail if we've really got the wrong
+                    // library, that's not a big problem. This is only called once per function in the native library.
+                    symbolName = "_" + symbolName + "@";
```

Comment:
See my comment, this is an issue with the .dll being built without a proper .def file to specify the actual exports. This should be done at the point of .dll production not consumption. These names are too fragile and (as you point out) too difficult to guess.